javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistAnnotationMemberDeclaration.java

```diff
@@ -88,16 +86,15 @@ public Expression getDefaultValue() {
     @Override
     public ResolvedType getType() {
         try {
-            CtClass returnType = annotationMember.getReturnType();
-            if (returnType.isPrimitive()) {
-                return ResolvedPrimitiveType.byName(returnType.getName());
-            }
-            SymbolReference<ResolvedReferenceTypeDeclaration> rrtd = typeSolver.tryToSolveType(returnType.getName());
-            if (rrtd.isSolved()) {
-                return new ReferenceTypeImpl(rrtd.getCorrespondingDeclaration(), typeSolver);
+            String descriptor = annotationMember.getMethodInfo().getDescriptor();
+            SignatureAttribute.MethodSignature signature = SignatureAttribute.toMethodSignature(descriptor);
+            SymbolReference<ResolvedReferenceTypeDeclaration> returnType = typeSolver.tryToSolveType(signature.getReturnType().jvmTypeName());
+            if (returnType.isSolved()) {
+                return new ReferenceTypeImpl(returnType.getCorrespondingDeclaration(), typeSolver);
```

Comment:
I think this change has nothing to do with this PR.
Perhaps you should create a new PR for it?

Reply:
I had to tweak the JavassistAnnotationMemberDeclaration to support the change, otherwise it will not resolve JRE types.
But if you prefer i can move this fix to another PR.

Reply:
ok no problem to leave it here in this PR.